src/content/datachannel/filetransfer/js/main.js

```diff
@@ -180,56 +182,69 @@ function receiveChannelCallback(event) {
   }
 }
 
-function onReceiveMessageCallback(event) {
-  console.log(`Received Message ${event.data.byteLength}`);
-  receiveBuffer.push(event.data);
-  receivedSize += event.data.byteLength;
+async function onReceiveMessageCallback(event) {
+  if (sendChannel && receiveChannel) {
```

Comment:
this is odd since you're using state only known at the sender on the receiving end?

Reply:
Ah, good spot, I left this when trying to figure why onerror is fired after the datachannel is closed, the only error type that should be thrown at close time is network error which is not the case here (even though close transport is network related, it's not an error per see and also of the wrong type)